RearrangingCarsTest.java

```diff
@@ -90,6 +92,32 @@ public void testDifferentLengthInputs() {
         Assert.assertNull(moves);
     }
 
+    @org.junit.Test
+    public void testOptimisedRearrange() {
+        for (int size = 5; size < 50; size++) {
+            int[] originalOrder = IntStream.rangeClosed(0, size - 1).toArray();
+            int[] finalOrder = originalOrder.clone();
+            shuffleArray(finalOrder);
```

Comment:
I had no idea! Actually, I remember having had problems with such statements before.. Anyways, it's good to know Java has this :)